OrbitCore/LinuxUtils.cpp

```diff
@@ -125,6 +126,18 @@ ErrorMessageOr<std::vector<ModuleInfo>> ListModules(int32_t pid) {
   return result;
 }
 
+ErrorMessageOr<std::vector<TracepointInfo>> ListTracepoints() {
+  std::vector<TracepointInfo> result;
+  std::string path = "/sys/kernel/debug/tracing/events";
+  for (const auto& entry : fs::directory_iterator(path)) {
+    TracepointInfo tracepoint_info;
+    tracepoint_info.set_name(fs::path(entry).filename());
+    result.emplace_back(tracepoint_info);
+  }
+
+  return result;
```

Comment:
If I am not mistaken this gives you list of categories but not complete list of tracepoints. For every directory you need do ls again to get all tracepoints in the given category.